app/js/actions/artifacts.js

```diff
@@ -53,7 +53,8 @@ export const createArtifact = (formData) => {
         const { connection: { uri, secretKey } } = getState();
         fetchAPI(secretKey, 'POST', `http://${uri}/api/workspace/artifacts`, formData)
         .catch(({ message: error }) => alert(error))
-        .then(() => dispatch(actions.refreshArtifacts()));
+        .then(() => dispatch(actions.refreshArtifacts()))
+        .then(() => dispatch(actions.checkImportableTypes()));
```

Comment:
Remove this line as it will dispatch your action twice (since `refreshArtifacts` already does this).